frontend/ui/uimanager.lua

```diff
@@ -714,7 +707,9 @@ function UIManager:handleInput()
 
     -- delegate input_event to handler
     if input_event then
-        self:_resetAutoSuspendTimer()
+        if input_event.handler ~= "onInputError" then
+            self:broadcastEvent(Event:new("InputEvent"))
```

Comment:
just throwing out one potential solution here. when we register the hook, we can also modify the close handler at runtime to have it deregister the hook before running the actual handler. for example, we can wrap it with a new function that does the clean before calling the original handler.